### PR TITLE
Release/1.0: Release candidate 0

### PR DESCRIPTION
**Pull request by anxolin**
`git log origin/master..release/1.0 --oneline --decorate=0`

**IMPORTANT**: Do not SQUASH this Merge. Use a regular merge to `master`. Same thing for the merge from `master` to `develop` 

## Summary
* TCR: For list of tokens
* Manage tokens
* Wrapping Ether
* Config file
* Override components
* Many fixes and improvements:

## Details:
b8611f8 Ignore all custom files (#1024)
1fb0a31 Use absolute path for importing components (#1023)
ccf3d97 Weth api, hooks (#852)
2b67526 Refactor spinners into a new component (#1025)
8d4c2f5 Double claim modal style (#1021)
4e05573 Manage tokens (#981)
ee90e54 Updated README with details for Config options (#1005)
95b789a 978/config file (#987)
ea991d3 Fix tcr override (#1004)
17ab5c4 944/enable token label (#1003)
df633dc Unhack Promise.all TS (#1002)
53492eb 180/tcr (#998)
9b6217e 960/dynamic token fetching (#967)
15307f3 More robust token id fetching (#951)
fd7b6cf Misc fixes (TokenSelector + styling) (#959)
2ab9302 948/update orders instead of replacing (#949)
3062623 Misc fixes (#945)
9af3f71 Alternative source (#947)
1562e0f 797/improve orders state (#930)
b7a690d 918/url encode token symbols (#924)
1715b30 LegalBanner: change skull font to icon (#934)
2fadde4 Use production URL for order book (#940)
9b22c86 Fix stale tokens (#929)
dbc8183 Add hops to order book (#939)
f180e34 switch to literal regexp (#933)
7c77194 Adjusting the wording as requested (#923)
50d2ae1 Merge branch 'master' into develop
347a4c4 only resubmit on mainnet (#927)

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr1043--dexreact.review.gnosisdev.com

File: config-default.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+#######################################################################################
+##                                  DEFAULT CONFIG                                   ##
+#######################################################################################
+# - Defines the default configuration
+# - **WARNING**: Do not modify this file
+#   - For custom config create a new file "custom/config.yaml"
+#   - You can override in the custom file any config that is defined here
+#   - For redefining components, or for more information, follow the instructions in ./README.md
+
+# TODO: Rename Dapp into Gnosis Protocol DApp
```
Comment (anxolin) [inline config-default.yaml:10]: Ups, this part was not implemented yet: https://github.com/gnosis/dex-react/issues/1045

File: package.json [hunk 2]
```diff
@@ -61,7 +29,7 @@
     "@fortawesome/free-solid-svg-icons": "^5.12.0",
     "@fortawesome/react-fontawesome": "^0.1.8",
     "@gnosis.pm/dapp-ui": "^0.5.3",
-    "@gnosis.pm/dex-js": "0.2.2",
+    "@gnosis.pm/dex-js": "0.3.0-RC.0",
```
Comment (anxolin) [inline package.json:32]: @alfetopito @W3stside should we close a version of dex-js?

File: src/api/weth/WethApiMock.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
+import { logDebug } from 'utils'
```
Comment (anxolin) [inline src/api/weth/WethApiMock.ts:1]: This is still pending https://github.com/gnosis/dex-react/issues/1044

File: config-default.yaml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+name: Mesa - Gnosis Protocol DApp
```
[the remark this answers is not in the document]
Comment (anxolin) [reply config-default.yaml:11]: The app still has many Mesa everywhere. I left that as a reminder to remove them all. I've made a similar comment and opened an issue so we address it in this version: https://github.com/gnosis/dex-react/issues/1045

File: config-default.yaml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+logoPath: './src/assets/img/logo.svg'
```
[the remark this answers is not in the document]
Comment (anxolin) [reply config-default.yaml:13]: Here: https://github.com/gnosis/dex-react/issues/1045

I added:
"Also, we should bring back the GP that we had. I think it's still in the project."

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (anxolin) [reply package.json:32]: Sure, although @W3stside hasn't bring the formatSmart yet, and maybe he'll need to create a version for it. So probably he'll do it

File: src/api/erc20/Erc20Api.ts [hunk 6]
```diff
@@ -7,8 +7,20 @@ import { TxOptionalParams, Receipt } from 'types'
 import { ZERO } from 'const'
 import { toBN } from 'utils'
 
+import ERC20_DETAILS from './erc20Details.json'
```
[the remark this answers is not in the document]
Comment (anxolin) [reply src/api/erc20/Erc20Api.ts:10]: Good point. I only touched components, but I think it's great to use absolute for these

However, let's not commit directly to this branch, and instead make PRs to it.

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (alfetopito) [reply package.json:32]: Hmmm, Dima already released a 0.3.1-RC version which might be incompatible/require further changes that I have in my branch

File: config-default.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (alfetopito) [reply config-default.yaml:13]: That's easy then:

```suggestion
logoPath: './src/assets/img/logo-df.svg'
```

Ops, that's the `dF` logo. Don't recall we ever having a `GP` logo.

File: src/api/erc20/Erc20Api.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (anxolin) [reply src/api/erc20/Erc20Api.ts:10]: I'm doing a PR now with your comments

File: package.json (hunk 2 above)
Comment (W3stside) [reply package.json:32]: wondering if @Velenir wants to include his latest PR inside the latest version

[the remark this answers is not in the document]
Comment (W3stside) [reply package.json:32]: I think we release a `0.3.1` that includes both my `formatSmart` and @Velenir 's new contract types via typechain

[the remark this answers is not in the document]
Comment (anxolin) [reply package.json:32]: Ideally if it's incompatible we can split it in 2 versions 0.4 and 0.5

File: config-default.yaml (hunk 1 above)
Comment (anxolin) [reply config-default.yaml]: Done!

File: package.json (hunk 2 above)
Comment (anxolin) [reply package.json]: I think you are generating 0.4.0 and PR'in it to `release/1.0`, so solving this issue...

**Pull request merged**